### PR TITLE
Sign compare warning fixes batch 9

### PR DESCRIPTION
**Pull request by tg-at-google**
in resolution of [-Wsign-compare] warnings with ids:
[
231, 237, 289, 291, 292, 304, 384, 
387, 389, 390, 391, 392, 393, 394, 
385, 395, 398, 399, 400, 407, 478, 
519
]

@mihaimaruseac

### CODE REVIEW

File: tensorflow/compiler/tf2xla/mlir_tf2xla.cc [hunk 1]
```diff
@@ -140,7 +140,7 @@ Status ConvertGraphDefToXlaViaMlir(
     if (!debug_info_path_begin_marker.empty()) {
       for (size_t i = 0, e = debug_info.files_size(); i < e; ++i) {
         std::string* file_name = debug_info.mutable_files(i);
-        size_t location =
+        int location =
             file_name->rfind(std::string(debug_info_path_begin_marker));
         if (location != -1) {
```
Comment (mihaimaruseac) [inline tensorflow/compiler/tf2xla/mlir_tf2xla.cc:145]: Replacing `-1` with [`std::string::npos`](https://en.cppreference.com/w/cpp/string/basic_string/npos) is better for readability

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

**Review by mihaimaruseac — APPROVED** [review]

**Comment by mihaimaruseac** [comment]
```
tensorflow/compiler/mlir/tensorflow/transforms/promote_resources_to_args.cc:345:36: error: 'eturn_operands' was not declared in this scope
   const int return_operands_size = eturn_operands.size();
                                    ^~~~~~~~~~~~~~
tensorflow/compiler/mlir/tensorflow/transforms/promote_resources_to_args.cc:345:36: note: suggested alternative: 'return_operands'
   const int return_operands_size = eturn_operands.size();
                                    ^~~~~~~~~~~~~~
                                    return_operands
```

**Comment by tg-at-google** [comment]
> ```
> tensorflow/compiler/mlir/tensorflow/transforms/promote_resources_to_args.cc:345:36: error: 'eturn_operands' was not declared in this scope
>    const int return_operands_size = eturn_operands.size();
>                                     ^~~~~~~~~~~~~~
> tensorflow/compiler/mlir/tensorflow/transforms/promote_resources_to_args.cc:345:36: note: suggested alternative: 'return_operands'
>    const int return_operands_size = eturn_operands.size();
>                                     ^~~~~~~~~~~~~~
>                                     return_operands
> ```

This is a file it seems I did not change.

**Comment by mihaimaruseac** [comment]
Right, it was a transient error fixed in the meanwhile

**Pull request merged**